### PR TITLE
Fix String escaping: Change escape character from : to % and fix functional tests

### LINKED ISSUE #92: Streaming calls with string fields in messages fail to parse
**Describe the bug**
As soon as streamed message contains a string field as last element in the message, parse fails

**To Reproduce**
```
gwhisper 127.0.0.1 examples.StreamingRpcs bidirectionalStreamEchoNumberOrString :str=asd: :str=asd2:
Parse failed.Parsed until: '127.0.0.1 examples.StreamingRpcs bidirectionalStreamEchoNumberOrString :str=asd :'
```

**Another more complex example which is probably caused by same issue:**
```
message FileHeader {
  string scope = 1; // name of the scope
  string path = 2; // path to the file
  string contentType = 3; // content type of the file
  int32 contentLength = 4; // content length of the file. Use -1 if the length is unknown.
}
message FilePart {
  bytes data = 1; // one part of the file data
}
message UploadFileRequest {
  oneof requestType {
    FileHeader header = 1; // meta data of the file
    FilePart filePart = 2; // part of the file data
  }
}
```
```
gwhisper localhost:41087 persistence.objectstore.ObjectStore uploadFile :header=:scope=test contentType=plain/text contentLength=5 path=/myfolder/myfile:: :filePart=:data=abcde::

Parse failed.Parsed until: 'localhost:41087 persistence.objectstore.ObjectStore uploadFile'
```

### PR DESCRIPTION
**Pull request by rainerschoe**
Fixes #92

### CODE REVIEW

**Comment by rainerschoe** [comment]
somehow github messed up and does not show correct diff

File: src/libCli/GrammarConstruction.cpp [hunk 1]
```diff
@@ -186,21 +186,7 @@ class GrammarInjectorMethodArgs : public GrammarInjector
...
-                        //auto stringContainer = m_grammar.createElement<Concatenation>("StringContainer");
-                        //stringContainer->addChild(m_grammar.createElement<FixedString>(":"));
-                        //stringContainer->addChild(m_grammar.createElement<RegEx>("[^:]*", "FieldValue"));
-                        //stringContainer->addChild(m_grammar.createElement<FixedString>(":"));
-                        //f_fieldGrammar->addChild(stringContainer);
-
-                        // Using this as a workaround until parser gets better regex support
-                        //f_fieldGrammar->addChild(m_grammar.createElement<RegEx>("[^:, ]*", "FieldValue"));
-                        f_fieldGrammar->addChild(m_grammar.createElement<EscapedString>(":, ", ':', "FieldValue"));
+                        f_fieldGrammar->addChild(m_grammar.createElement<EscapedString>(":, %", '%', "FieldValue"));
```
Comment (anna-riesch) [inline src/libCli/GrammarConstruction.cpp:189]: A short summary of each of the cases might help to make it easier to understand the addFieldValueGrammar function. Also there is a small typo in l. 118. General question: Why do some comments start with more than two slashes?

File: tests/testServer/ServiceStreamingRpcs.cpp [hunk 2]
```diff
@@ -103,3 +103,16 @@ ::grpc::Status ServiceStreamingRpcs::bidirectionalStreamNegateNumbers(
...
+            ::grpc::ServerContext* context,
+            ::grpc::ServerReaderWriter< ::examples::NumberOrStringOneOf,
+            ::examples::NumberOrStringOneOf>* stream
+            )
+{
+    ::examples::NumberOrStringOneOf message;
+    while (stream->Read(&message)) {
+        stream->Write(message);
+    }
+    return grpc::Status();
```
Comment (anna-riesch) [inline tests/testServer/ServiceStreamingRpcs.cpp:117]: What Status would be returned here?

**Review by anna-riesch — COMMENTED** [review]
Some questions before merging.

File: src/libCli/GrammarConstruction.cpp (hunk 1 above)
Comment (rainerschoe) [reply src/libCli/GrammarConstruction.cpp:202]: I added a new commit with improved documentation.

this is "doxygen-style" documentation:

two slashes "//" denote a comment intended for developers working on the code to help understand how the code works and how it is implemented.

three slashes "///" denote documentation of an interface of a function or class which is important for people using the code (without them needing to understand how it is implemented)

In short:
Always use three slashes when you document the function declaration itself, two when you document something inside a function

File: tests/testServer/ServiceStreamingRpcs.cpp (hunk 2 above)
Comment (rainerschoe) [reply tests/testServer/ServiceStreamingRpcs.cpp:117]: Default constructed status is a "Good" status

File: src/libCli/GrammarConstruction.cpp [hunk 3]
```diff
@@ -115,36 +115,45 @@ class GrammarInjectorMethodArgs : public GrammarInjector
...
 
-        /// Construct Grammar for a protocol buffer field value and add it to existing grammae.
+        /// Construct Grammar for a protocol buffer field value and add it to existing grammar.
         /// @param f_fieldGrammar Grammar element to which the field grammar will be added as a child.
         /// @param f_field protobuf field descriptor representing the field.
         /// @param f_maxRecursionDepth max number of nested messages. If this is 0, no grammar for sub-messages is added anymore if the field is of message type. Instead a fixed String is added to stop recursion.
         void addFieldValueGrammar(GrammarElement * f_fieldGrammar, const grpc::protobuf::FieldDescriptor * f_field, size_t f_maxRecursionDepth)
         {
 
+            // Adding grammar for a specofic protobuf field.
```
Comment (anna-riesch) [inline src/libCli/GrammarConstruction.cpp:125]: Small Typo

**Review by anna-riesch — COMMENTED** [review]
Found a typo. Otherwise everything else looks good.

**Review by anna-riesch — APPROVED** [review]
